worker.go

```diff
@@ -145,26 +145,30 @@ func detectContent(imageFormat, name, path string, headers map[string]string, pa
 	return
 }
 
+// detectNamespaces returns a list of unique namespaces detected in a layer and its ancestry.
 func detectNamespaces(name string, files tarutil.FilesMap, parent *database.Layer) (namespaces []database.Namespace, err error) {
-	namespaces, err = featurens.Detect(files)
+	nsSet := map[string]*database.Namespace{}
+	nsCurrent, err := featurens.Detect(files) // detect the namespaces of current layer's files
```

Comment:
no need for this comment; trailing comments in Go are usually discouraged anyways